app/client-v2/src/stores/projectStore.ts

```diff
@@ -75,50 +76,76 @@ export const useProjectStore = defineStore("project", {
     },
 
     async processFiles(files: File[]) {
-      for (const file of files) {
-        const content = await file.text();
-        const fileHash = Md5.hashStr(content);
-        this.project.samples.push({ hash: fileHash, filename: file.name });
+      this.uploadingPercentage = 0;
+      const hashedFileBatches = await this.batchFilesForProcessing(files);
+      this.processFileBatches(hashedFileBatches);
+    },
+    async batchFilesForProcessing(files: File[]) {
+      const batchSize = 5; // found to be best balance of throughput and memory usage
+      const hashedFiles: HashedFile[] = await Promise.all(
+        files.map(async (file) => {
+          const content = await file.text();
+          const fileHash = Md5.hashStr(content);
+          return { hash: fileHash, filename: file.name, file };
+        })
+      );
+      const hashedFileBatches: HashedFile[][] = [];
+      for (let i = 0; i < hashedFiles.length; i += batchSize) {
+        hashedFileBatches.push(hashedFiles.slice(i, i + batchSize));
+      }
+      return hashedFileBatches;
+    },
```

Comment:
yeah it gets bad too with lots of files!!! i tried to see if quick fix but couldn't find it.. thus have created this [ticket](https://mrc-ide.myjetbrains.com/youtrack/issue/bacpop-177/Fix-Uploading-initial-wait) where it will be fixed